Improve handling of translated menu accelerators

vadz opened the pull request:
Fixes #24323 and replaces #25130.

vadz committed:
Fix dangling references to temporaries in internat sample

In some build variants (e.g. when wxUSE_INTL==0) _() is a macro doing
nothing, in particular not involving a function call, and the temporary
wxString created by it doesn't survive past the end of the statement and
hence a reference to it can't be used.

Just make a copy of the string to fix this.

vadz committed:
Don't bother using non-existent translation of "RawCtrl"

As this modifier is never shown to the user, it never needs to be
translated, so don't bother checking if we have its translation.

vadz committed:
Translate "RawCtrl" just as "Ctrl"

The string "RawCtrl" should never appear in the UI, so use it only for
the internal accelerator representation, but translate this modifier as
"Ctrl" for the end users.

vadz committed:
Don't translate "+" separating modifier from accelerator

Look up translations of "Alt", "Ctrl" and "Shift" instead of the same
strings with "+" after it, as this makes more sense and allows to reuse
the existing translations for these strings.

vadz committed:
Use translations of "Ctrl" rather than "ctrl"

And the same for "Alt" and "Shift".

This is done both because the capitalized forms are more common and
because this will allow reusing these translations, whose case doesn't
matter as they're only used inside a case-insensitive comparison, in
other places, where the case should be correct.

vadz committed:
Capitalize "Ctrl", "Shift" and "Alt" message IDs

After the last commit, these strings need to be translated rather than
the lowercase versions used before, so update all the message catalogs
to use them.

Also capitalize the translations for the languages for which it was easy
to do so.

And remove "rawctrl" translations that are not used.

vadz committed:
Improve documentation of translating menu accelerators

Explain that wxWidgets message catalogs are supposed to contain the
translations of the modifiers keys, but may not have them for all
languages.

Also document that these translations use "keyboard key" context.

This replaces #25130.

vadz committed:
Avoid crash if wxMenuItemAccelEntry unit test fails

Use REQUIRE, not CHECK, before dereferencing a possibly null pointer.

taler21 commented on docs/doxygen/overviews/internationalization.h:
```diff
@@ -111,13 +111,18 @@ It is of course possible to use the Unix layout everywhere instead.
+modifiers in the "keyboard key" context must be provided. The wxWidgets own
+translations do provide them for many languages, but not for all of them, so if
+you notice that the accelerators don't work when translated, please provide the
+translations for these strings in your own context or, even better, contribute
```
What do you mean by `in your own context`?
Shouldn't that be better `in your own catalogs`?
The context must be "keyboard key", regardless of whether the translations are provided by wxWidgets or the application itself.

vadz replied:
Yes, this is a typo, thanks for noticing it, it should have been "your own catalogs".

vadz committed:
Restore correct modifiers order in menu accelerators under MSW

The changes of 4dcda62f9d (Normalize displayed accelerators in wxMSW
too, 2022-12-04) accidentally switched the order in which modifiers were
displayed in menu accelerators in wxMSW.

Fix this by reordering code in wxAcceleratorEntry to return the
accelerator text with the modifiers in MSW native order. Note that other
platforms don't use the same order, but they don't use this function
either, so this change doesn't affect them.
